src/processing/DataNormalizer.py

```diff
@@ -30,3 +31,6 @@ def explode_column_nyt(df,column:str='keywords'):
     exploded_df = exploded_df.drop(column, axis=1).join(tmp)
 
     return exploded_df
+
+def remove_punctuation(text):
```

Comment:
I miss type hinting and docstrings here and some other places. These things are not an obligation, but if you choose to write them, it's nice to keep the style consistent across the whole codebase